alicloud/resource_alicloud_cms_sitemonitor.go

```diff
@@ -0,0 +1,193 @@
+package alicloud
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cms"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudCmsSiteMonitor() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudCmsSiteMonitorCreate,
+		Read:   resourceAlicloudCmsSiteMonitorRead,
+		Update: resourceAlicloudCmsSiteMonitorUpdate,
+		Delete: resourceAlicloudCmsSiteMonitorDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"address": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"task_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"task_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{SiteMonitorHTTP, SiteMonitorDNS, SiteMonitorFTP, SiteMonitorPOP3,
+					SiteMonitorPing, SiteMonitorSMTP, SiteMonitorTCP, SiteMonitorUDP}, false),
+			},
+			"alert_ids": {
+				Type:     schema.TypeString,
```

Comment:
If the `alert_ids` is a list, please set it to `TypeList`

Reply:
It is defined and imlemented as simple string. please see https://help.aliyun.com/document_detail/115048.html

Reply:
The http request does not support send list data, so the api parameter type is string. But, this parameter stands a list of id and it should be set to TypeList or TypeSet.

Reply:
done